registry/storage/driver/swift/swift.go

```diff
@@ -628,13 +607,28 @@ func (d *driver) Delete(ctx context.Context, path string) error {
 			}
 			return err
 		}
+	}
 
-		if err := d.Conn.ObjectDelete(d.Container, obj.Name); err != nil {
-			if err == swift.ObjectNotFound {
-				return storagedriver.PathNotFoundError{Path: obj.Name}
+	if d.BulkDeleteSupport && len(objects) > 0 {
+		filenames := make([]string, len(objects))
+		for i, obj := range objects {
+			filenames[i] = obj.Name
+		}
+		if _, err := d.Conn.BulkDelete(d.Container, filenames); err != swift.Forbidden {
+			if err == swift.ContainerNotFound {
+				return storagedriver.PathNotFoundError{Path: path}
 			}
 			return err
```

Comment:
We fall back only if BulkDelete returns swift.Forbidden, which means that bulk delete is not supported.
But as we use bulk delete only if support for it has been detected, it's probably not useful.